### ISSUE TITLE
eDNAjoint: an R package for interpreting paired environmental DNA and traditional surveys

### ISSUE THREAD

**Issue by abigailkeller** [ropensci/software-review#628]
Submitting Author Name: Abigail Keller
Submitting Author Github Handle: @abigailkeller
Repository:  https://github.com/abigailkeller/eDNAjoint
Submission type: Pre-submission
Language: en

---

-   Paste the full DESCRIPTION file inside a code block below:

```
Version: 0.1
Maintainer: Abigail G. Keller <agkeller@berkeley.edu>
Authors@R: 
    c(person("Abigail G.", "Keller", role = c("aut", "cre"), email="agkeller@berkeley.edu"),
    person("Ryan P.", "Kelly", role = "ctb", email="rpkelly@uw.edu"))
Description: Models integrate environmental DNA (eDNA) detection data and traditional survey data to jointly estimate species catch rate. Models can be used with count data via traditional survey methods (i.e., trapping, electrofishing, visual) and replicated eDNA detection/nondetection data via polymerase chain reaction (i.e., PCR or qPCR) from multiple survey locations. Estimated parameters include probability of a false positive eDNA detection, a site-level covariates that scale the sensitivity of eDNA surveys relative to traditional surveys, and catchability coefficients for traditional gear types. Models are implemented with a Bayesian framework (Markov chain Monte Carlo) using the 'Stan' probabilistic programming language.
License: GPL-3
Encoding: UTF-8
Roxygen: list(markdown = TRUE, roclets = c ("namespace", "rd", "srr::srr_stats_roclet"))
RoxygenNote: 7.3.1
Biarch: true
Depends: 
    R (>= 3.4.0)
Imports: 
    bayestestR,
    dplyr,
    ggplot2,
    loo,
    magrittr,
    methods,
    Rcpp (>= 0.12.0),
    RcppParallel (>= 5.0.1),
    rstan (>= 2.26.23),
    rstantools (>= 2.3.1.1),
    tidyr
LinkingTo: 
    BH (>= 1.66.0),
    Rcpp (>= 0.12.0),
    RcppEigen (>= 0.3.3.3.0),
    RcppParallel (>= 5.0.1),
    rstan (>= 2.26.23),
    StanHeaders (>= 2.26.22)
SystemRequirements: GNU make
LazyData: true
Suggests: 
    knitr,
    rmarkdown,
    testthat (>= 3.0.0)
VignetteBuilder: knitr
Config/testthat/edition: 3
```

## Scope 

- Please indicate which category or categories from our [package fit policies](https://ropensci.github.io/dev_guide/policies.html#package-categories) or [statistical package categories](https://stats-devguide.ropensci.org/overview.html#overview-categories) this package falls under. (Please check an appropriate box below):

    **Data Lifecycle Packages**

	- [ ] data retrieval
	- [ ] data extraction
	- [ ] data munging
	- [ ] data deposition
    - [ ] data validation and testing
	- [ ] workflow automation
	- [ ] version control
	- [ ] citation management and bibliometrics
	- [ ] scientific software wrappers
	- [ ] field and lab reproducibility tools
	- [ ] database software bindings
	- [ ] geospatial data
	- [ ] text analysis
    
     **Statistical Packages**

	- [x] Bayesian and Monte Carlo Routines
	- [ ] Dimensionality Reduction, Clustering, and Unsupervised Learning
	- [ ] Machine Learning
	- [ ] Regression and Supervised Learning
	- [ ] Exploratory Data Analysis (EDA) and Summary Statistics
	- [ ] Spatial Analyses
	- [ ] Time Series Analyses

- Explain how and why the package falls under these categories (briefly, 1-2 sentences).  Please note any areas you are unsure of:

This package runs Bayesian statistical models written in Stan using data from paired environmental DNA (eDNA) and traditional surveys provided by a user. Additional functions support interpretation of model output.

- If submitting a statistical package, have you already [incorporated documentation of standards into your code via the **srr** package](https://stats-devguide.ropensci.org/pkgdev.html#pkgdev-srr)?

Not yet.

-   Who is the target audience and what are scientific applications of this package?  

The target audience is environmental science researchers or managers who want to interpret environmental DNA data but do not have experience writing and implementing custom Bayesian models.

-   Are there other R packages that accomplish the same thing? If so, how does yours differ or meet [our criteria for best-in-category](https://ropensci.github.io/dev_guide/policies.html#overlap)?

To my knowledge, no other R package implements similar types of models. Although the R package contains additional model variations, the model was originally developed in a publication: "Tracking an invasion front with environmental DNA", led by the authors of this submission.

-   (If applicable) Does your package comply with our [guidance around _Ethics, Data Privacy and Human Subjects Research_](https://devguide.ropensci.org/policies.html#ethics-data-privacy-and-human-subjects-research)?

Yes

-  Any other questions or issues we should be aware of?:

**Comment by ldecicco-USGS** [comment]
Hi @abigailkeller 
I'm new to the editor process for the stats submissions, so sorry for the delay.

The stat submission guide is here:
https://stats-devguide.ropensci.org/pkgdev.html

I think the next step for you is to follow the directions here:
https://stats-devguide.ropensci.org/pkgdev.html#pkgdev-srr
https://github.com/ropensci-review-tools/srr

The eDNAjoint would need srr results to >50% before we can move the package to a full submission.

**Comment by abigailkeller** [comment]
Closing issue as my package seems out of scope for statistical submissions

**Comment by mpadge** [comment]
Hi @abigailkeller, Mark here from the rOpenSci stats team, to politely disagree with you about your package being out of scope. Scope decisions for stats packages are pretty straightforward: If you think that your package can comply with at least half of all applicable standards - in your case [General](https://stats-devguide.ropensci.org/standards.html#standards-general) and [Bayesian](https://stats-devguide.ropensci.org/standards.html#standards-bayesian), then it is by definition in scope.

Please feel free to have a look through your code, documentation, and tests, check all standards you think will likely apply, and if that seems like more than half, you are very welcome to open this issue again. Then, once you've documented compliance, ask the bot to call `check srr` yourself. As long as that requirement is met, the submission may then proceed. Thanks :smile:

**Comment by abigailkeller** [comment]
Re-opened issue. I believe my package is now passing at least 55% of the general statistics and Bayesian statistics standards. Thanks!

**Comment by ldecicco-USGS** [comment]
@ropensci-review-bot check srr

**Comment by mpadge** [comment]
@abigailkeller Sorry, check results got lost due to upgrades in our system. They'll appear sometime tomorrow. Sorry for any inconvenience.

**Comment by abigailkeller** [comment]
Just to add to the ropensci-review-bot:
I believe the missing standards (G1.5, G5.4, G5.5,BS1.2b, BS1.2c, BS1.4, BS4.0, BS4.3, BS6.2, BS6.3) are in the .Rmd file in the vignette folder

**Comment by mpadge** [comment]
@ldecicco-USGS That `srr` check by the bot should now be correct. @abigailkeller You can call [`srr::srr_stats_pre_submit()`](https://docs.ropensci.org/srr/reference/srr_stats_pre_submit.html) locally to confirm that your package will pass, and then ask the bot here to `check srr` when you think you've finished documenting. Sorry both for any inconvenience here.

**Comment by abigailkeller** [comment]
check srr

**Comment by ldecicco-USGS** [comment]
Thank you @abigailkeller for this pre-submission inquiry! This package is well within our statistical scope and we'd be happy to review a full submission. So now you'll need to open a new issue for the full submission. Let me know if you have any questions. 

(see issue 615 for a recent example of a submission, where the pre-submission is 614)
